aten/src/ATen/native/mps/MPSFallback.cpp

```diff
@@ -0,0 +1,208 @@
+//  Copyright © 2022 Apple Inc.
+
+#include <ATen/ATen.h>
+#include <ATen/native/mps/Copy.h>
+#include <ATen/core/dispatch/Dispatcher.h>
+#include <ATen/core/op_registration/op_registration.h>
+#include <ATen/native/MathBitsFallback.h>
+#include <ATen/core/ivalue.h>
+#include <ATen/core/stack.h>
+#include <ATen/core/boxing/KernelFunction.h>
+#include <ATen/core/dispatch/Dispatcher.h>
+#include <torch/library.h>
+#include <ATen/Functions.h>
+
+namespace at {
+
+struct MPSFallback {
+  MPSFallback() : key(DispatchKey::MPS), op_name("mps") {}
+  std::vector<at::Tensor> to_cpu(const at::TensorList& tensors) {
+      // We can't just call at::to_cpu() on the entire list of Tensors
+      // Because it will break on undefined tensors. Separate out undefined tensors first.
+      std::vector<at::Tensor> cpu_tensors(tensors.size());
+      std::vector<at::Tensor> valid_tensors;
+      std::vector<bool> to_translate(tensors.size());
+      for (const auto i : c10::irange(tensors.size())) {
+          const at::Tensor& tensor = tensors[i];
+          // Explicitly handling undefined tensors here instead of letting `at::_to_cpu` handle it.
+          // Otherwise, we'd need to require all backends with their own implementation of _to_cpu
+          // to properly handle undefined tensors.
+          if (tensor.defined()) {
+              to_translate[i] = true;
+              valid_tensors.push_back(tensor);
+          } else {
+              cpu_tensors[i] = tensor;
+          }
+      }
+      auto cpu_valid_tensors = at::_to_cpu(valid_tensors);
+      for (size_t i = 0, defined_pos = 0; i < tensors.size(); ++i) {
+          if (to_translate[i]) {
+              cpu_tensors[i] = std::move(cpu_valid_tensors[defined_pos++]);
+          }
+      }
+    return cpu_tensors;
+  }
+  c10::optional<c10::Device> compute_target_device(std::vector<at::Tensor>& t_args, std::vector<c10::List<at::Tensor>> tlist_args) {
```

Comment:
nit: this should always be mps no?

Reply:
Yeah actually I am not using this function, so removed it. It is indeed only `kMPS`